amgut/db/patches/0029.sql

```diff
@@ -0,0 +1,121 @@
+ALTER TABLE barcodes.plate ADD finalized_timestamp timestamp  ;
+ALTER TABLE barcodes.plate ADD CONSTRAINT plate_pkey PRIMARY KEY ( barcode ) ;
+
+CREATE INDEX idx_plate ON barcodes.plate ( project_id ) ;
+CREATE INDEX idx_plate_0 ON barcodes.plate ( ext_robot ) ;
+CREATE INDEX idx_plate_1 ON barcodes.plate ( kf_robot ) ;
+CREATE INDEX idx_plate_2 ON barcodes.plate ( ext_kit_lot ) ;
+CREATE INDEX idx_plate_3 ON barcodes.plate ( protocol ) ;
+CREATE INDEX idx_plate_4 ON barcodes.plate ( person ) ;
+ALTER TABLE barcodes.plate ADD CONSTRAINT fk_plate FOREIGN KEY ( barcode ) REFERENCES barcodes.barcode( barcode )    ;
+ALTER TABLE barcodes.plate ADD CONSTRAINT fk_plate_0 FOREIGN KEY ( project_id ) REFERENCES barcodes.project( project_id )    ;
+ALTER TABLE barcodes.plate ADD CONSTRAINT fk_plate_1 FOREIGN KEY ( ext_robot ) REFERENCES barcodes.robots( robot_id )    ;
+ALTER TABLE barcodes.plate ADD CONSTRAINT fk_plate_2 FOREIGN KEY ( kf_robot ) REFERENCES barcodes.robots( robot_id )    ;
+ALTER TABLE barcodes.plate ADD CONSTRAINT fk_plate_3 FOREIGN KEY ( ext_kit_lot ) REFERENCES barcodes.lot( lot_id )    ;
+ALTER TABLE barcodes.plate ADD CONSTRAINT fk_plate_4 FOREIGN KEY ( protocol ) REFERENCES barcodes.protocol( protocol_id )    ;
+ALTER TABLE barcodes.plate ADD CONSTRAINT fk_plate_5 FOREIGN KEY ( person ) REFERENCES barcodes.person( person_id )    ;
+
+CREATE TABLE barcodes.lot ( 
+	lot_id               bigserial  NOT NULL,
+	lot_type             varchar  NOT NULL,
+	lot                  varchar  NOT NULL,
+	date_added           date DEFAULT current_date NOT NULL,
+	CONSTRAINT pk_lots PRIMARY KEY ( lot_id )
+ ) ;
+
+CREATE TABLE barcodes.person ( 
+	person_id            bigint  NOT NULL,
+	name                 varchar(100)  NOT NULL,
+	email                varchar  NOT NULL,
+	date_added           date DEFAULT current_date NOT NULL,
+	CONSTRAINT pk_person PRIMARY KEY ( person_id )
+ ) ;
+
+CREATE TABLE barcodes.plate_sample ( 
+	barcode              bigint  NOT NULL,
+	well                 varchar(3)  NOT NULL,
+	sample               varchar  NOT NULL,
+	CONSTRAINT pkey_plate_barcode_1 PRIMARY KEY ( barcode, well )
+ ) ;
+
+CREATE INDEX idx_plate_barcode ON barcodes.plate_sample ( well ) ;
+
+CREATE INDEX idx_plate_barcode_0 ON barcodes.plate_sample ( barcode ) ;
+
+CREATE TABLE barcodes.protocol ( 
+	protocol_id          bigserial  NOT NULL,
+	protocol_name        varchar(100)  NOT NULL,
+	template_dna         integer  NOT NULL,
+	CONSTRAINT pk_protocol PRIMARY KEY ( protocol_id )
+ ) ;
+
+COMMENT ON COLUMN barcodes.protocol.template_dna IS 'Template DNA added, in uL';
+
+CREATE TABLE barcodes.robots ( 
+	robot_id             bigserial  NOT NULL,
+	robot_type           varchar  NOT NULL,
+	robot_name           varchar(100)  NOT NULL,
+	date_added           date DEFAULT current_date NOT NULL,
+	CONSTRAINT pk_robots PRIMARY KEY ( robot_id )
+ ) ;
+
+CREATE TABLE barcodes.pcr_plate ( 
```

Comment:
I discourage the usage of quoted columns names, it can add complexity on the python code...